##### Null Hypothese H0: The new bus line will not shorten the commuting time (T-new >=T-old)
##### Alternative Hypothese H1: The new bus line will shorten the commuting time (T-new < T-old)
###### I will use a significance level $\alpha=0.05$,which means I want the probability of getting a result at least as significant as mine to be less then 5%

In [41]:
import pandas as pd
import numpy as np

In [42]:
!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt > bus.csv

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0   7738      0 --:--:-- --:--:-- --:--:--  7705


In [43]:
P_mean = 36
P_stv = 6
size=bus.count
np.random.normal(P_mean,P_stv,size=100)


array([ 39.18149987,  24.74354454,  34.76797847,  39.67853394,
        35.22111348,  34.81463989,  44.75995935,  37.28168191,
        31.7362634 ,  38.65783556,  44.36514943,  30.0551474 ,
        37.54905794,  43.85148796,  37.0742895 ,  31.21483552,
        33.78725649,  39.11382212,  29.88729098,  34.51531569,
        38.40687494,  31.76886796,  41.66846529,  37.27285264,
        40.99624507,  25.89264901,  25.05751543,  43.12012774,
        38.3195725 ,  31.63074792,  30.64560552,  44.8789561 ,
        33.1941904 ,  32.43698635,  24.68062899,  30.1907087 ,
        38.58725508,  37.47344067,  30.42999794,  37.7923096 ,
        25.74413439,  28.4277024 ,  42.82998152,  33.05512847,
        36.3805697 ,  31.9590414 ,  22.83445956,  36.16382871,
        23.1512335 ,  40.08107323,  32.50225496,  36.15142039,
        33.02163219,  43.4789448 ,  32.18994559,  27.81925657,
        44.98639656,  43.54624268,  36.11646479,  41.1411957 ,
        28.46263723,  36.55640982,  37.283353  ,  32.45

In [44]:
bus=pd.read_csv("bus.csv",header=None)

In [45]:
bus.describe()

0
count  100.000000
mean    34.466162
std      7.137280
min     16.527267
25%     30.020542
50%     34.693057
75%     39.603456
max     48.749877

In [46]:
S_mean=bus.mean()

In [47]:
MeanDef=P_mean-S_mean
Lower=P_stv/100**0.5
z_score=MeanDef/Lower

In [48]:
print(z_score)

0    2.556397
dtype: float64


##### The Z-score we get is 2.56, which is larger than 1.96. It is out of the 2-sigma range, thus we can reject the null hypothese. The new bus line will shorten the commuting time